### PR TITLE
V8.2.2 merge

### PR DESCRIPTION
**Pull request by jderrico-noaa**
Re submitting the v8.2.2 code merge

### CODE REVIEW

File: src/core_atmosphere/physics/mpas_atmphys_driver.F [hunk 1]
```diff
@@ -260,23 +267,23 @@ subroutine physics_driver(domain,itimestep,xtime_s)
...
        call deallocate_sfclayer(block%configs)
     endif
 
     !call to 1d ocean mixed-layer model
-    if(config_oml1d) call driver_oml1d(block%configs, mesh, diag, diag_physics, sfc_input)
+    if(config_oml1d) call driver_oml1d(block%configs,mesh,diag,diag_physics,sfc_input)
 
     !call to land-surface scheme:
-    if(config_lsm_scheme .ne. 'off') then
+    if(config_lsm_scheme .eq. 'sf_noah' .or. config_lsm_scheme .eq. 'sf_ruc') then
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F:280]: we should check to make sure we aren't missing a "sf_noahmp" option.

**Review by joeolson42 — APPROVED** [review]
I only looked at 34 files, but I don't see any duplicate lines and the key files look as I expected.

**Comment by jderrico-noaa** [comment]
This should be identical to the previous 8.2.2 branch, I think the problems were in the v8.2 merge, which I rolled back and am hoping this will now compile/run

**Review by mdtoyNOAA — APPROVED** [review]

**Review by tanyasmirnova — APPROVED** [review]

**Comment by barlage** [comment]
I ran two tests: mesoscale_reference and mesoscale_reference+bl_mynn both using 71791d535 in 8.2.2_merge compared to edca469f from joe's fixes_20241024 branch from last week. The results are probably OK to merge, but we have some issues that will need to be sorted out. There are differences in diagnostics, which I assume are expected but since both of these use Noah LSM (and ncar suites in general), I don't think we want any changes since this will definitely affect backward compatibility. I'm OK with punting on these for now just to get things in gsl/develop so we can start addressing them.
```
nccmp -dsSqf jnt-71791d535-mesorefmynn.mesoscale_reference.gsl.conus.gfs.2023031015/history.2023-03-10_16.00.00.nc joe-prprepost-mesorefmynn.mesoscale_reference.gsl.conus.gfs.2023031015/history.2023-03-10_16.00.00.nc
Variable Group Count       Sum    AbsSum         Min       Max     Range       Mean     StdDev
q2       /         5 0.0213712 0.0213712 0.000784278 0.0138528 0.0130686 0.00427423 0.00555739
t2m      /         5   1.77234   1.77234   0.0128479   1.43073   1.41788   0.354468   0.603874
th2m     /         5   1.77588   1.77588   0.0128479   1.43405    1.4212   0.355176   0.605334

nccmp -dsSqf jnt-71791d535-mesoref.mesoscale_reference.gsl.conus.gfs.2023031015/history.2023-03-10_16.00.00.nc joe-prprepost-mesoref.mesoscale_reference.gsl.conus.gfs.2023031015/[history.2023-03-10_16.00.00.nc
Variable Group Count          Sum      AbsSum          Min         Max       Range         Mean      StdDev
q2       /         4 -0.000224113 0.000368446 -0.000175074 7.14448e-05 0.000246519 -5.60281e-05 0.000112383
t2m      /         3     0.483795    0.774811    -0.145508     0.57959    0.725098     0.161265    0.375199
th2m     /         3     0.484894     0.77652    -0.145813    0.580963    0.726776     0.161631    0.376085
```

**Review by barlage — APPROVED** [review]

**Pull request merged**